ulauncher/modes/extensions/extension_socket_server.py

```diff
@@ -83,10 +86,11 @@ def handle_registration(self, framer: JSONFramer, event: dict[str, Any]) -> None
                 for msg_id in pended[1:]:
                     GObject.signal_handler_disconnect(framer, msg_id)
             ext_id: str | None = event.get("ext_id")
+            path: str | None = event.get("path")
             assert ext_id
+            assert path
             ExtensionSocketController(self.socket_controllers, framer, ext_id)
-            # TODO: This is ugly, but we have no other way to detect the extension started successfully
-            ExtensionController.create(ext_id).is_running = True
+            self.on_extension_registered(ext_id, Path(path))
```

Comment:
I don't think this is better than before. Sure, it looks better here, but you have the exact same logic in a method in the extension mode now instead, and you need extra logic to pass that method here.

The fundamental problem is needing ot set this property from the outside and it's still the same. Plus this adds more conflicts for #1534 where I actually did fix this proper (and I still haven't given up on that PR, I have kept rebasing it and improving it and trying to replace with socket pairs now)